### PR TITLE
treat ambient non-aliased 'require' as commonjs 'require'

### PR DESCRIPTION
**Pull request by vladima**
fixes https://github.com/Microsoft/TypeScript/issues/11752

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -12819,16 +12819,41 @@ namespace ts {
...
-                // Make sure require is not a local function
-                !resolveName(node.expression, (<Identifier>node.expression).text, SymbolFlags.Value, /*nameNotFoundMessage*/ undefined, /*nameArg*/ undefined)) {
+            if (isInJavaScriptFile(node) && isCommonJsRequire(node)) {
                 return resolveExternalModuleTypeByLiteral(<StringLiteral>node.arguments[0]);
             }
 
             return getReturnTypeOfSignature(signature);
         }
 
+        function isCommonJsRequire(node: Node) {
```
Comment (sheetalkamat) [inline src/compiler/checker.ts]: May be we need to use this function in completions.ts to determine whether to give module names as completion list?

Comment (sheetalkamat) [inline src/compiler/checker.ts]: May be we need to use this function in completions.ts to determine whether to give module names as completion list?

Comment (sheetalkamat) [inline src/compiler/checker.ts]: May be we need to use this function in completions.ts to determine whether to give module names as completion list?

Comment (vladima) [inline src/compiler/checker.ts]: certainly but it is not related to the original issue so I'd put it into different PR

Comment (mhegazy) [inline src/compiler/checker.ts]: +1
